source/engines/toastEngine.js

```diff
@@ -0,0 +1,158 @@
+	}
+
+	// Make database entities
+	const recentToasts = await database.models.Toast.findAll({ where: { companyId: interaction.guildId, senderId: interaction.user.id, createdAt: { [Op.gt]: new Date(new Date() - 2 * timeConversion(1, "d", "ms")) } }, include: database.models.Toast.Recipients });
+	let rewardsAvailable = 10;
+	let critToastsAvailable = 2;
+	for (const toast of recentToasts) {
+		for (const reciept of toast.Recipients) {
+			if (reciept.isRewarded) {
+				rewardsAvailable--;
+			}
+			if (reciept.wasCrit) {
+				critToastsAvailable--;
+			}
+		}
+	}
+	const toastsInLastDay = recentToasts.filter(toast => new Date(toast.createdAt) > new Date(new Date() - timeConversion(1, "d", "ms")));
+	const hunterIdsToastedInLastDay = toastsInLastDay.reduce((idSet, toast) => {
+		toast.Recipients.forEach(reciept => {
+			if (!idSet.has(reciept.recipientId)) {
+				idSet.add(reciept.recipientId);
+			}
+		})
+		return idSet;
+	}, new Set());
+
+	const lastFiveToasts = await database.models.Toast.findAll({ where: { companyId: interaction.guildId, senderId: interaction.user.id }, include: database.models.Toast.Recipients, order: [["createdAt", "DESC"]], limit: 5 });
+	const staleToastees = lastFiveToasts.reduce((list, toast) => {
+		return list.concat(toast.Recipients.filter(reciept => reciept.isRewarded).map(reciept => reciept.recipientId));
+	}, []);
+
+	const rawRecipients = [];
+	const rewardedRecipients = [];
+	let critValue = 0;
+	const [season] = await database.models.Season.findOrCreate({ where: { companyId: interaction.guildId, isCurrentSeason: true } });
+	season.increment("toastsRaised");
+	await database.models.User.findOrCreate({ where: { id: interaction.user.id } });
+	const [sender] = await database.models.Hunter.findOrCreate({ where: { userId: interaction.user.id, companyId: interaction.guildId } });
+	sender.increment("toastsRaised");
+	const toast = await database.models.Toast.create({ companyId: interaction.guildId, senderId: interaction.user.id, text: toastText, imageURL });
+	for (const id of toasteeIds) {
+		//TODO #97 move to bulkCreate after finding solution to create by association only if user doesn't already exist
+		const [user] = await database.models.User.findOrCreate({ where: { id } });
+		const rawToast = { toastId: toast.id, recipientId: id, isRewarded: !hunterIdsToastedInLastDay.has(id) && rewardsAvailable > 0, wasCrit: false };
+		if (rawToast.isRewarded) {
+			rewardedRecipients.push(id);
+
+			// Calculate crit
+			if (critToastsAvailable > 0) {
+				const critRoll = Math.random() * 100;
+
+				let effectiveToastLevel = sender.level + 2;
+				for (const recipientId of staleToastees) {
+					if (id == recipientId) {
+						effectiveToastLevel--;
+						if (effectiveToastLevel < 2) {
+							break;
+						}
+					}
+				};
+
+				// f(x) = 150/(x+2)^(1/3)
+				if (critRoll * critRoll * critRoll > 3375000 / effectiveToastLevel) {
+					rawToast.wasCrit = true;
+					critValue += 1;
+					critToastsAvailable--;
+				}
```

Comment:
The math behind critical toasting is odd and abstract enough that the calculation may be worth a private function (in the future).